pallets/corporate-actions/src/distribution/mod.rs

```diff
@@ -503,17 +529,13 @@ impl<T: Trait> Module<T> {
         <Portfolio<T>>::unlock_tokens(&dist.from, &dist.currency, &amount)
     }
 
-    // Compute `balance * amount / supply`, i.e. DID's benefit.
-    fn benefit_of(
-        balance: T::Balance,
-        amount: T::Balance,
-        supply: T::Balance,
-    ) -> Result<T::Balance, DispatchError> {
+    // Compute `balance * per_share`, i.e. DID's benefit.
+    fn benefit_of(balance: T::Balance, per_share: T::Balance) -> Result<T::Balance, DispatchError> {
         balance
-            .checked_mul(&amount)
-            .ok_or(Error::<T>::BalanceAmountProductOverflowed)?
-            .checked_div(&supply)
-            .ok_or_else(|| Error::<T>::BalanceAmountProductSupplyDivisionFailed.into())
+            .checked_mul(&per_share)
+            // `per_share` was entered as a multiple of 1_000_000.
+            .map(|v| v / T::Balance::from(1_000_000u32))
```

Comment:
Rather than inlining 10^6 we should specify it as a constant and use that in both the pallet and the runtime tests.